fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/ResolveOrdinalInOrderByAndGroupBy.java

```diff
@@ -0,0 +1,96 @@
+                            List<OrderKey> orderKeys = sortNode.getOrderKeys();
+                            List<OrderKey> orderKeysWithoutConst = new ArrayList<>();
+                            for (OrderKey k : orderKeys) {
+                                Expression expression = k.getExpr();
+                                expression = FoldConstantRule.INSTANCE.rewrite(expression);
+                                if (expression instanceof IntegerLikeLiteral) {
+                                    IntegerLikeLiteral i = (IntegerLikeLiteral) expression;
+                                    int ord = i.getIntValue();
+                                    checkOrd(ord, childOutput.size());
+                                    orderKeysWithoutConst
+                                            .add(new OrderKey(childOutput.get(ord - 1), k.isAsc(), k.isNullFirst()));
+                                } else {
+                                    orderKeysWithoutConst.add(k);
+                                }
+                            }
+                            return sortNode.withOrderByKey(orderKeysWithoutConst);
+                        })
+                ))
+                .add(RuleType.ResolveOrdinalInGroupBy.build(
+                        logicalAggregate().then(agg -> {
+                            List<NamedExpression> aggOutput = agg.getOutputExpressions();
+                            List<Expression> groupByWithoutConst = new ArrayList<>();
+                            for (Expression groupByExpr : agg.getGroupByExpressions()) {
+                                groupByExpr = FoldConstantRule.INSTANCE.rewrite(groupByExpr);
+                                if (groupByExpr instanceof IntegerLikeLiteral) {
+                                    IntegerLikeLiteral i = (IntegerLikeLiteral) groupByExpr;
+                                    int ord = i.getIntValue();
+                                    Expression aggExpr = aggOutput.get(ord - 1);
+                                    if (!CollectionUtils.isEmpty(aggExpr.children())
+                                            && aggExpr.child(0) instanceof Literal) {
+                                        continue;
+                                    }
+                                    checkOrd(ord, aggOutput.size());
+                                    groupByWithoutConst.add(aggExpr);
+                                } else {
+                                    groupByWithoutConst.add(groupByExpr);
+                                }
+                            }
+                            return new LogicalAggregate(groupByWithoutConst, agg.getOutputExpressions(), agg.child());
+                        }))).build();
+    }
+
+    private void checkOrd(int ord, int childOutputSize) {
+        if (ord < 1 || ord > childOutputSize) {
+            throw new IllegalStateException(String.format("INVALID ORD: %s", ord));
```

Comment:
add Rule Name in  exception message please.
it help us find where goes wrong by log.